src/Symfony/Component/HttpFoundation/ChainRequestMatcher.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpFoundation;
+
+/**
+ * ChainRequestMatcher compares a pre-defined set of checks against a Request instance.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class ChainRequestMatcher implements RequestMatcherInterface
+{
+    /**
+     * @param RequestMatcherInterface[] $matchers
+     */
+    public function __construct(private array $matchers = [])
+    {
+    }
+
+    /**
+     * @return $this
+     */
+    public function add(RequestMatcherInterface $matcher): static
```

Comment:
Do we even need this `add` ? Without this method, we could make the constructor accept an iterable and use IteratorArgument. (not sure it is worth it as core matchers have no dependencies so lazy-loading them won't bring much, but who knows about future ones...)